### DISCUSSION TITLE
Issue while booting up the application .

### DISCUSSION THREAD

**Discussion by praveen2211** [OktopUSP/oktopus#96] [Q&A] [answered]
@leandrofars  post execution of docker compose command images(mongo,mochi,golang node etc) are up but when we try execute 
**"go run main.go -u root -P root -mongo "mongodb://127.0.0.1:27017"** .

I am facing the below error in my local environement could you please help me on the same.
Seems like application is expecting an properties file .

/oktopus/backend/services/controller/cmd/oktopus$ go run main.go -u root -P root -mongo "mongodb://127.0.0.1:27017"
2023/08/02 14:40:13 Loaded variables from '.env'
2023/08/02 14:40:13 Error to load environment variables: open .env: no such file or directory
2023/08/02 14:40:13 main.go:46: Starting Oktopus Project TR-369 Controller Version: 0.0.1
2023/08/02 14:40:13 db.go:23: Trying to ping Mongo database...
2023/08/02 14:40:43 db.go:26: Couldn't connect to MongoDB --> server selection error: server selection timeout, current topology: { Type: Unknown, Servers: [{ Addr: 127.0.0.1:27017, Type: Unknown, Last error: dial tcp 127.0.0.1:27017: connect: connection refused }, ] }
exit status 1

**Comment by leandrofars** [comment]
Controller is trying to login into a mongodb in your localhost, which doesn't seem to exist. Try to use a valid mongo url.

**Comment by leandrofars** [comment]
You can use this docker compose to start a mongodb instance in your machine: 
<pre>
version: '3.8'  # You can use a different version if required.

services:
  mongodb:
    image: mongo:latest  # You can specify a specific version here if needed.
    container_name: my_mongodb
    ports:
      - "27017:27017"
</pre>
Run with 'docker-compose up'

**Reply by praveen2211** [reply]
@leandrofars images of mongodb,redis,go lang etc are up but while i try to bring up application through main.go it is expecting environment variables
also where should we add this configuration ?

**Reply by leandrofars** [reply, marked as answer]
You might use a .env.local from the directory you are running the controller. If you go to 'oktopus/backend/services/controller' you'll see there is a .env file, you can copy this file to a .env.local in the same folder and fill the 'SECRET_API_KEY' inside of it, and then from that folder you execute: 'go run cmd/oktopus/main.go  -u root -P root -mongo "mongodb://127.0.0.1:27017" '. This SECRET_API_KEY  variable is responsable for signed jwts, if you're running a local app you don't need this env variable, but it's nice to have in production. 
![image](https://github.com/leandrofars/oktopus/assets/83298718/904dc03f-d779-43ae-bc99-ad1573294ad5)
